staging/src/k8s.io/code-generator/cmd/status-gen/status-generators/status.go

```diff
@@ -0,0 +1,442 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status_generators
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/gengo/args"
+	"k8s.io/gengo/examples/set-gen/sets"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+
+	"k8s.io/klog"
+)
+
+// CustomArgs is used tby the go2idl framework to pass args specific to this generator.
+type CustomArgs struct {
+}
+
+// This is the comment tag that carries parameters for API status generation.  Because the cadence is fixed, we can predict
+// with near certainty when this lifecycle happens as the API is introduced.
+const (
+	tagEnabledName    = "k8s:status-gen"
+	introducedTagName = tagEnabledName + ":introduced"
+	deprecatedTagName = tagEnabledName + ":deprecated"
+	removedTagName    = tagEnabledName + ":removed"
```

Comment:
do these always refer to kubernetes-apiserver versions? since this is put in the generic code-generator repo, trying to understand how it would interact with the following setups:
1. other API servers building other types outside kube-apiserver and specifying version numbers unrelated to kube
2. API servers serving kube types in a server with a different view of its own version number

Reply:
> do these always refer to kubernetes-apiserver versions? since this is put in the generic code-generator repo, trying to understand how it would interact with the following setups:
> 
> 1. other API servers building other types outside kube-apiserver and specifying version numbers unrelated to kube
> 2. API servers serving kube types in a server with a different view of its own version number

Generator is opt-in, but if someone chose to opt-in, they would use their own version numbers, though major.minor is fixed.  Similar to kube, they would have to provide version.Get results to make decisions.